src/libraries/Common/tests/System/Runtime/Serialization/Utils.cs

```diff
@@ -149,7 +149,7 @@ private static bool CompareXElements(XElement baselineXElement, XElement actualX
         {
             // Check whether the XName is the same, this can be done by comparing the two XNames.
 
-            if (!baselineXElement.Name.Equals(actualXElement.Name))
+            if ((baselineXElement == null && actualXElement != null) || !baselineXElement.Name.Equals(actualXElement.Name))
```

Comment:
Are you trying to protect against nulls being passed in? If so, this can still throw NRE. If you pass in null for both, the first expression of the OR will be false, and then actualXElement.Name will throw NRE.